### DISCUSSION TITLE
Does smartdns support DOH3?

### DISCUSSION THREAD

**Discussion by Mile-Lile** [pymumu/smartdns#1700] [Q&A] [answered]
When I enter [link of my doh3 dns provider](https://doh3.dns.nextdns.io) in my Chrome browser (chrome://settings/security - Select DNS provider) on [nextdns test page](https://test.nextdns.io/) i get result like this:
![Screenshot from 2024-03-14 22-26-52](https://github.com/pymumu/smartdns/assets/12752816/c889daf0-5d54-4c54-a650-db6bb1f054e2)
but when I enter `server-https https://doh3.dns.nextdns.io` in my smartdns configuration, I see that I am using http2 (tls 1.2):

![Screenshot from 2024-03-14 22-25-01](https://github.com/pymumu/smartdns/assets/12752816/db3b610b-386c-4d4f-80e2-a01fa203fec5)

So my question is, how to setup https3 (DOH3) on smartdns?

**Comment by Mile-Lile** [comment]
just to add, there are some open-source libraries that support http/3:

[lsquic](https://github.com/litespeedtech/lsquic)
[nghttp3](https://github.com/ngtcp2/nghttp3)
[libcurl](https://github.com/curl/curl)
[MsQuic](https://github.com/microsoft/msquic)
[quiche](https://github.com/cloudflare/quiche)
[Cronet](https://github.com/chromium/chromium/tree/main/net/quic)

**Comment by pymumu** [comment, marked as answer]
It is not supported yet because openssl's quic support is not yet complete.

**Comment by wangmice** [comment]
现在支持doh3了吗？

**Reply by lalasou** [reply]
> 现在支持doh3了吗？

现在支持了 
H3  quic好像都支持了

**Comment by Mile-Lile** [comment]
I can tell you that DOH3 is working on OpenWRT with [luci-app-https-dns-proxy](https://docs.openwrt.melmac.net/https-dns-proxy/#HTTP3QUICSupport) installed with modified out-of-tree fork of OpenSSL

and on

[ControlD](https://github.com/Control-D-Inc/ctrld)

those I tested with NextDNS...
